openapi/src/main/java/io/micronaut/openapi/swagger/ObjectMapperFactory.java

```diff
@@ -0,0 +1,243 @@
+import io.micronaut.openapi.swagger.jackson.mixin.OpenAPI31Mixin;
+import io.micronaut.openapi.swagger.jackson.mixin.OpenAPIMixin;
+import io.micronaut.openapi.swagger.jackson.mixin.OperationMixin;
+import io.micronaut.openapi.swagger.jackson.mixin.Schema31Mixin;
+import io.micronaut.openapi.swagger.jackson.mixin.SchemaMixin;
+import io.micronaut.openapi.swagger.jackson.util.DeserializationModule;
+import io.micronaut.openapi.swagger.jackson.util.DeserializationModule31;
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.ExternalDocumentation;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.callbacks.Callback;
+import io.swagger.v3.oas.models.examples.Example;
+import io.swagger.v3.oas.models.headers.Header;
+import io.swagger.v3.oas.models.info.Contact;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.info.License;
+import io.swagger.v3.oas.models.links.Link;
+import io.swagger.v3.oas.models.links.LinkParameter;
+import io.swagger.v3.oas.models.media.DateSchema;
+import io.swagger.v3.oas.models.media.Discriminator;
+import io.swagger.v3.oas.models.media.Encoding;
+import io.swagger.v3.oas.models.media.EncodingProperty;
+import io.swagger.v3.oas.models.media.MediaType;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.media.XML;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponse;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import io.swagger.v3.oas.models.security.OAuthFlow;
+import io.swagger.v3.oas.models.security.OAuthFlows;
+import io.swagger.v3.oas.models.security.Scopes;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import io.swagger.v3.oas.models.servers.Server;
+import io.swagger.v3.oas.models.servers.ServerVariable;
+import io.swagger.v3.oas.models.servers.ServerVariables;
+import io.swagger.v3.oas.models.tags.Tag;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationConfig;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.BeanSerializerModifier;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.Feature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+public class ObjectMapperFactory {
```

Comment:
All classes in package - just copy from swagger-core. Here are the classes for serialization/deserialization. How to be in this case? I'm not sure if we need to describe javadocs for them or anything. It might be worth making a note at the package level that these classes are taken from the swagger core

Reply:
@graemerocher All classes in package - just copy from swagger-core. Here are the classes for serialization/deserialization. How to be in this case? I'm not sure if we need to describe javadocs for them or anything. It might be worth making a note at the package level that these classes are taken from the swagger core

Reply:
Ok in that case mark them as internal at least with `@since` tags

Reply:
@graemerocher I added this javadoc:
```java
/**
 * This class is copied from swagger-core library.
 *
 * @since 4.6.0
 */
```
Is it ok?